omod/src/main/webapp/resources/scripts/custom/deletePatient.js

```diff
@@ -0,0 +1,66 @@
+var delPatient = delPatient || {};
+var jqObject = $();
+
+delPatient.deletePatientCreationDialog = null;
+
+/**
+ * Functions used to Delete a Patient
+ */
+
+delPatient.returnUrl = "/coreapps/findpatient/findPatient.page?app=coreapps.findPatient";
+
+delPatient.createDeletePatientCreationDialog = function() {
+    delPatient.deletePatientCreationDialog = emr.setupConfirmationDialog({
+        selector: '#delete-patient-creation-dialog',
+        actions: {
+            confirm: function() {
+                var deleteReason = jQuery('#delete-reason').val().trim(); //Retrieve text from text box
+                if(deleteReason && deleteReason.length > 0) { //Should not be invalid or empty
+                    jq.ajax({
+                        url: '/openmrs/ws/rest/v1/patient/'+delPatient.patientUUID,
+                        type: 'DELETE',
+                        body: {
+                            "void_reason": deleteReason
+                        },
+                        success: function() {
+                            jqObject.toastmessage( 'showToast', { type: 'success',
+                                position: 'top-right',
+                                text:  emr.successMessage("Success!") } );
+                            jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
+                            delPatient.deletePatientCreationDialog.close();
+                            jq(setTimeout(delPatient.goToReturnUrl, 1300)); //Allow the success message to display before redirecting to another page
+                        },
+                        error: function() {
+                            jqObject.toastmessage( 'showToast', { type: 'error',
```

Comment:
You also need to do more debugging when you run into errors and look at the error messages more closely, the JS error that says 'coreapps is not defined' is a clue that JS things coreapps is a variable since you didn't put the code in quotes